src/main.rs

```diff
@@ -32,32 +35,61 @@ impl Terminal {
 
     fn should_ask_for_todo(&mut self) -> bool {
         let mut buf = String::new();
-        self.stdin.read_line(&mut buf).unwrap();
+
+        if let Err(_) = self.stdin.read_line(&mut buf) {
+            self.show_error(Err(TerminalError::StdoutErr));
+        }
 
         buf.trim() == "s"
     }
 
     fn ask_for_new_todo(&mut self) -> Todo {          
 
         println!("\nVocê gostaria de adicionar um novo TODO? 🤔 (Digite: 's' para SIM ou qualquer outra tecla para NÃO)");
-        
-        let response = self.should_ask_for_todo();
-        
-        if !response {
+                
+        if !self.should_ask_for_todo() {
             println!("\nTodo list finalizado! 🤠");
             std::process::exit(0);            
         } 
 
         println!("\nQual TODO deseja criar?");
 
         let mut new_todo = String::new();
-        self.stdin.read_line(&mut new_todo).unwrap();
+
+        if let Err(_) = self.stdin.read_line(&mut new_todo) {
+            self.show_error(Err(TerminalError::StdoutErr));
+        }
+
         let todo_message = new_todo.trim().to_string();
 
         Todo { message: todo_message }
     }
 
-    fn show_todo(&mut self, todo: &Todo) {        
-        writeln!(self.stdout, "\n✅: {}", todo.message).unwrap();
+    fn show_todo(&mut self, todo: &Todo) -> Result<(), io::Error> { 
+        writeln!(self.stdout, "\n✅: {}", todo.message)?;
+        Ok(())
+    }
+
+    fn show_error(&mut self, data: Result<(), TerminalError>) {
+
+        if let Err(error) = data {
+            println!("{}", TerminalError::message_err(error));
+        }
+       
+    }
+}
+
+enum TerminalError {    
+    WriteErr,
+    StdoutErr,
+}
+
+impl TerminalError {
+
+    fn message_err(msg: TerminalError) -> String {
```

Comment:
```suggestion
    fn message_err(self) -> String {
```

Dá para usar o `self` aqui! Aí, em vez de chamar `TerminalError::message_err(error)`, você pode fazer apenas `error.message_err()`.

Reply:
Boa! Vou alterar isso.